test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +248,83 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases converts release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	scheme, host, err := goproxy.GetSchemeAndHost(os.Getenv("GOPROXY"))
+	if err != nil {
+		return err
+	}
+	if scheme == "" || host == "" {
+		return fmt.Errorf("releasemarker does not support disabling the go proxy: GOPROXY=%q", os.Getenv("GOPROXY"))
+	}
+	goproxyClient := goproxy.NewClient(scheme, host)
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type == "url" {
+				releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+				if !strings.EqualFold(version.Name, releaseMarker) {
```

Comment:
Can we have some godocs here how that works?

Maybe instead of using EqualFold we could use:


```suggestion
				if strings.HasPrefix(version.Name, "}") && strings.HasSuffix(version.Name, "{") {
          // Skip version because it is not a resolvable marker. Resolvable markers are surrounded by `{}`
          continue
        }
        releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
```